lib/dartino/launch_dartino.dart

```diff
@@ -0,0 +1,121 @@
+import 'dart:async';
+
+import 'package:atom/node/fs.dart';
+import 'package:atom/node/process.dart';
+import 'package:logging/logging.dart';
+
+import '../atom.dart';
+import '../launch/launch.dart';
+import '../projects.dart';
+import '../state.dart';
+
+final Logger _logger = new Logger('atom.dartino_launch');
+
+class DartinoLaunchType extends LaunchType {
+  static void register(LaunchManager manager) =>
+      manager.registerLaunchType(new DartinoLaunchType());
+
+  DartinoLaunch _lastLaunch;
+
+  DartinoLaunchType() : super('dartino');
+
+  bool canLaunch(String path) {
+    DartProject project = projectManager.getProjectFor(path);
+    if (project == null || !project.isDartinoProject()) return false;
+    if (!analysisServer.isExecutable(path)) return false;
+    return true;
+  }
+
+  List<String> getLaunchablesFor(DartProject project) {
```

Comment:
OK.

There was a change that went into the analysis server a while ago that may be what you want. We can query to get the transitive imports of a library. It'll let the client decide what an application type a library is (web, flutter, dartino, ...) w/o having to add specific support for each type into the analysis server.